api/src/main/java/org/openmrs/module/fhir/api/strategies/condition/GenericConditionStrategy.java

```diff
@@ -20,14 +20,34 @@
 public interface GenericConditionStrategy {
 
 	/**
-	 * Get condition by id
+	 * Get condition by uuid
 	 *
 	 * @param uuid the uuid of the condition that need to retrieve
-	 * @return Condition Return fhir condition resource and will return null if condition is not found for the given id
+	 * @return Condition Return fhir condition resource and will return null if condition is not found for the given uuid
 	 */
 	Condition getConditionById(String uuid);
 
+	/**
+	 * Search condition by uuid
+	 *
+	 * @param uuid the uuid of the condition that need to retrieve
+	 * @return Condition Return a List of FHIR condition resource and will return Empty list if condition is not found for the given uuid
```

Comment:
Did you confirm that it actually returns an empty list instead of null?

Reply:
It's not yet implemented.
That's search method was editing what was there.
https://issues.openmrs.org/projects/FM/issues/FM-65 this will take care of that